xbmc/GUIPassword.cpp

```diff
@@ -479,6 +479,32 @@ bool CGUIPassword::CheckMenuLock(int iWindowID)
     return true;
 }
 
+bool CGUIPassword::IsVideoUnlocked()
+{
+  const auto profileManager = CServiceBroker::GetSettingsComponent()->GetProfileManager();
+
+  const bool isLocked{profileManager->GetCurrentProfile().videoLocked()};
+  if (!isLocked && !m_strMediaSourcePath.empty()) // check mediasource by path
+    return g_passwordManager.IsMediaPathUnlocked(profileManager, "video");
+
+  if (isLocked)
+    return IsMasterLockUnlocked(true); //Now let's check the PW if we need!
+  return true;
+}
+
+bool CGUIPassword::IsMusicUnlocked()
+{
+  const auto profileManager = CServiceBroker::GetSettingsComponent()->GetProfileManager();
+
+  const bool isLocked{profileManager->GetCurrentProfile().musicLocked()};
+  if (!isLocked && !m_strMediaSourcePath.empty()) // check mediasource by path
+    return g_passwordManager.IsMediaPathUnlocked(profileManager, "music");
+
+  if (isLocked)
+    return IsMasterLockUnlocked(true); //Now let's check the PW if we need!
+  return true;
+}
+
```

Comment:
this looks hardly like code duplication.
can we introduce a member `bool CGUIPassword::IsMediaUnlocked(const std::string& mediaType)` ?
instead of `MusicUnlocked()` and `VideoUnlocked()` .... then call it like `IsMediaUnlocked(MediaTypeVideo)` ?

Reply:
> this looks hardly like code duplication. can we introduce a member `bool CGUIPassword::IsMediaUnlocked(const std::string& mediaType)` ? instead of `MusicUnlocked()` and `VideoUnlocked()` .... then call it like `IsMediaUnlocked(MediaTypeVideo)` ?

Not sure that would save much code - still need to know if video or music as `.videoLocked()` or `.musicLocked()` as well as passing `"music"` or `"video"` to `IsMediaPathUnlocked()`

Reply:
right, and you don't have to. was only a suggestion ;)

Reply:
> right, and you don't have to. was only a suggestion ;)

Thanks, always appreciated :) Will make the other change to make it clearer.